astrocut/asdf_cutouts.py

```diff
@@ -16,19 +17,28 @@
 from astropy.modeling import models
 
 
-def _get_cloud_http(s3_uri: Union[str, S3Path]) -> str:
+def _get_cloud_http(s3_uri: Union[str, S3Path], verbose: bool = False) -> str:
     """ 
     Get the HTTP URI of a cloud resource from an S3 URI.
 
     Parameters
     ----------
     s3_uri : string | S3Path
         the S3 URI of the cloud resource
+    verbose : bool
+        Default False. If true intermediate information is printed.
     """
-    # create file system
-    fs = s3fs.S3FileSystem()
 
-    # open resource and get URL
+    # check if public or private by sending an HTTP request
+    s3_path = S3Path.from_uri(s3_uri) if isinstance(s3_uri, str) else s3_uri
+    url = f'https://{s3_path.bucket}.s3.amazonaws.com/{s3_path.key}'
+    resp = requests.head(url)
+    is_anon = False if resp.status_code == 403 else True
+    if verbose and not is_anon:
+        print(f'Attempting to access private S3 bucket: {s3_path.bucket}')
+
+    # create file system and get URL of file
+    fs = s3fs.S3FileSystem(anon=is_anon)
```

Comment:
This looks much cleaner to me and I like the approach here.  Does this add much overhead to the call?  I know `request.head` is a pretty lightweight request so I imagine it's negligible .  Do we need to add something in case this request fails or times out?  We could add a `timeout` [keyword](https://requests.readthedocs.io/en/latest/user/advanced/#timeouts) to the request.

Reply:
Yep, I think `requests.head` is a good choice here since it doesn't download any body content. Adding a timeout is a good call as well, I will push another commit!